alacritty_terminal/src/config/mod.rs

```diff
@@ -144,7 +143,7 @@ pub struct Config<T> {
 
 impl<T> Config<T> {
     #[inline]
-    pub fn draw_bold_text_with_bright_colors(&self) -> bool {
+    pub const fn draw_bold_text_with_bright_colors(&self) -> bool {
```

Comment:
Why did you change all of these to `const` functions?

Reply:
The lint commit is full of changes at the direction of the linter. The linter is correct here, those getter functions are const. The untouched getters operate on options, I'm not even sure if there is a feature gate for that.

Reply:
Interesting, I wasn't aware of this lint it must be new.

Of course these functions can be marked as `const`, but it's pointless. Could you point me to the exact clippy/Rust version you're running? Since the latest nightly that supports rustfmt and clippy does not warn on this.

Reply:
I try to go above and beyond when it comes to annoying the maintainers of software I use. `-W clippy::nursery`

Reply:
I'll add that `const` better expresses intent, and allows for `const` propagation:
https://rust-lang.github.io/rust-clippy/master/index.html#missing_const_for_fn

Reply:
> that const better expresses intent

It better expresses intent for what?

> allows for const propagation

We don't need that though, you've mentioned in your other comment that you'd like to simplify things and this seems to go in the exact opposite direction.

Adding `const` to all functions just because they could potentially maybe sometime be called from a `const` context doesn't really make much sense when you know that the chance for that is basically nill and they're only for internal use. If we need something to be `const` internally, we can just change it when we need it.

These are functions that read information from a configuration file that is deserialized. There is no usecase to do that in a `const` context. It's just verbose and I'd argue that it goes completely against expressing intent. If you make something `const` you're expressing the intent that it could potentially be interesting to call this from a const context and thus it should not be changed to a 'normal' function unless absolutely necessary. It will be a breaking change to do so and will be completely impossible if consumers already rely on that. I don't think this is what we're trying to do here? It seems like this change is mostly 'it currently does not hurt to make it const', which is completely different.

Reply:
> It better expresses intent for what?

Being a getter, providing no utility (aside from API boundary) beyond direct reference to the struct member. I am presupposing that they are getter functions though, and that they aren't intended to be stuffed with run time data manipulation logic.

> These are functions that read information from a configuration file that is deserialized.

These functions read information from a struct. That is not a distinction without a difference, especially when you start thinking about default values, where they are stored and how they are influenced by build time flags.

Reply:
> Being a getter, providing no utility (aside from API boundary) beyond direct reference to the struct member.

Who says we want to guarantee that? It's not a good idea to lock an API down to a restriction like that unless you're absolutely certain that's always going to be the case and you actually gain any benefits from doing so. Currently you're trying to lock things down without any benefits at all.

> These functions read information from a struct. That is not a distinction without a difference

It is pointless to look at code without looking at the bigger picture of where and how it is used. Only because it might sometimes be useful for struct access to be `const`, doesn't mean it should always be `const` whenever possible.

It makes no sense for this struct to be accessed in a `const` environment.

Reply:
> Who says we want to guarantee that? It's not a good idea to lock an API down...

Nobody, as far as I know. I didn't even know that they were there to guarantee anything.

> It is pointless to look at code without looking at the bigger picture of where and how it is used.

I agree, one can't look at the present state of the config code and conclude "yeah, this is a sane place to rely on for default values that can be overridden at run time."

Reply:
> this is a sane place to rely on for default values that can be overridden at run time

That's not an accurate description of what the configuration is for. It is entirely a runtime structure, so these aren't just default values that can be overriden at runtime, this is a runtime configuration structure that can change at any time during execution.